src/transformers/trainer.py

```diff
@@ -2330,6 +2342,9 @@ def evaluation_loop(
             # Prediction step
             loss, logits, labels = self.prediction_step(model, inputs, prediction_loss_only, ignore_keys=ignore_keys)
 
+            if is_torch_tpu_available():
+                xm.mark_step()
```

Comment:
This part is done by the dataloader (which is wrapped in a `ParallelLoader`), so it shouldn't be here.

Reply:
This is intentional since `loss` will be materialized in `self._nested_gather(loss.repeat(batch_size))` and adding a mark_step here can significantly improve the speed. For example, the evaluation time of `bert-base-uncased` using `run_mlm.py` will be reduced from 32.53s to 18.73s by adding this mark_step.